test/forms/change_note_form_test.rb

```diff
@@ -0,0 +1,53 @@
+require "test_helper"
+
+class ChangeNoteFormTest < ActiveSupport::TestCase
+  include ActionDispatch::TestProcess
+
+  test "is invalid when minor_change is not present" do
+    change_note_form = ChangeNoteForm.new(change_note: "This is present", minor_change: nil)
+    assert_not change_note_form.valid?
+  end
+
+  test "is invalid when it is a major change and change_note is not present" do
+    change_note_form = ChangeNoteForm.new(change_note: nil, minor_change: false)
+    assert_not change_note_form.valid?
+  end
+
+  test "is valid when it is a minor change and a change_note is not present" do
+    change_note_form = ChangeNoteForm.new(change_note: nil, minor_change: true)
+    assert change_note_form.valid?
+  end
+
+  test "is valid when it is a major change and change_note is present" do
+    change_note_form = ChangeNoteForm.new(change_note: "This is present", minor_change: false)
+    assert change_note_form.valid?
+  end
+
+  test "#save returns false and adds errors to the form object when invalid" do
+    change_note_form = ChangeNoteForm.new
+    edition = build(:edition)
+
+    assert_equal false, change_note_form.save!(edition)
+    assert_equal ["Minor change Select whether this is a major or minor update"], change_note_form.errors.full_messages
+  end
+
+  test "#save updates the edition successfully when it is a major change and a change note is present" do
+    edition = create(:edition)
```

Comment:
Potentially controversial, but what do you think of potentially stubbing the edition doing something like the [date validation test](https://github.com/alphagov/whitehall/blob/main/test/unit/app/models/date_validation_test.rb). I did a little bit of profiling of the Whitehall test suite using [StackProf](https://github.com/tmm1/stackprof) the other day and found out about 2/3 of the test suite run time is spent waiting on the database. So anything we can do to reduce the number of database calls in our tests could make a big impact.

Reply:
Correction, I used https://github.com/test-prof/test-prof, I think with the StackProf integration. Might have been the rubyprof one though. Anyway, it's a pretty cool tool

Reply:
I think build_stubbed won't work when it needs to interact with the database though right? Which it will for these tests

Reply:
Would it not be sufficient to check that `update!` was called with the right parameters? I thought `build_stubbed` did that but apparently not - but we could use our own stub to do it.

On the profiling side, I didn't check the specific directory but we easily could if you want to do a bit more poking around. Might be worth seeing if we can pair on that at some point

Reply:
Yea agreed. I must admit i'm always a bit way of heavy mocking/creating stubbed classes. But stubbing the update call seems like it's an excellent pattern to follow to me for unit tests

Reply:
Have updated the tests to use build_stubbed 👍